services/tables/src/main/java/com/linkedin/openhouse/tables/services/TablesServiceImpl.java

```diff
@@ -211,6 +219,18 @@ public List<AclPolicy> getAclPolicies(
     return authorizationHandler.listAclPolicies(tableDto, userPrincipal);
   }
 
+  @Override
+  public Optional<DataAccessCredential> getDataAccessCredential(
+      String databaseId, String tableId, Map<String, String> params) {
+    TableDto tableDto = getTableOrThrow(databaseId, tableId);
+    StorageType.Type tableStorageType =
+        new StorageType().fromString(URI.create(tableDto.getTableLocation()).getScheme());
+
+    Storage storage = storageManager.getStorage(tableStorageType);
+
+    return storage.getDataAccessCredentialForTableLocation(tableDto.getTableLocation(), params);
```

Comment:
Is the storage layer going to interact with STS service to return the STS Access token?